internal/server/singleprocess/service_instance.go

```diff
@@ -18,8 +21,43 @@ func (s *service) ListInstances(
 	var err error
 	switch scope := req.Scope.(type) {
 	case *pb.ListInstancesRequest_DeploymentId:
-		result, err = s.state.InstancesByDeployment(scope.DeploymentId, nil)
+		// For more complex deployments, sometimes it takes a bit longer for
+		// the instance to respond. This blocks on a response when querying
+		// a deployment by id, and lets the user know it is taking a while
+		//
+		// The default case is no timeout, no blocking. Make a request and return
+
+		timeout := "0"
+		if req.ConnectTimeout != "" {
+			timeout = req.ConnectTimeout
+		}
+		connectTimeout, err := time.ParseDuration(timeout)
+		if err != nil {
+			return nil, err
+		}
+
+		if connectTimeout > 0 {
```

Comment:
I think you can change the above timeout parsing stuff and change this if to `if req.ConnectTimeout !=""`